functions/New-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,341 @@
+                }
+
+                if ($OnSuccessAction) {
+                    Write-Message -Message "Setting job step success action to $OnSuccessAction" -Level Verbose 
+                    $JobStep.OnSuccessAction = $OnSuccessAction
+                }
+
+                if ($OnSuccessStepId) {
+                    Write-Message -Message "Setting job step success step id to $($OnSuccessStepId)" -Level Verbose 
+                    $JobStep.OnSuccessStep = $OnSuccessStepId
+                }
+
+                if ($OnFailAction) {
+                    Write-Message -Message "Setting job step fail action to $OnFailAction" -Level Verbose 
+                    $JobStep.OnFailAction = $OnFailAction
+                }
+
+                if ($OnFailStepId) {
+                    Write-Message -Message "Setting job step fail step id to $($OnFailStepId)" -Level Verbose 
+                    $JobStep.OnFailStep = $OnFailStepId
+                }
+
+                if ($DatabaseName) {
+                    # Check if the database is present on the server
+                    if (($Server.Databases).Name -contains $DatabaseName) {
+                        Write-Message -Message "Setting job step database name to $DatabaseName" -Level Verbose 
+                        $JobStep.DatabaseName = $DatabaseName
+                    }
+                    else {
+                        Stop-Function -Message ("The database is not present on instance $instance.") -Target $JobName
+                        return
+                    }
+                }
+
+                if ($DatabaseUserName -and $DatabaseName) {
+                    # Check if the username is present in the database
+                    if (($Server.Databases[$DatabaseName].Users).Name -contains $DatabaseUserName) {
+                    
+                        Write-Message -Message "Setting job step database username to $DatabaseUserName" -Level Verbose 
+                        $JobStep.DatabaseUserName = $DatabaseUserName
+                    }
+                    else {
+                        Stop-Function -Message ("The database user is not present in the database $DatabaseName on instance $instance.") -Target $StepName
+                        return
+                    }
+                }
+
+                if ($RetryAttempts) {
+                    Write-Message -Message "Setting job step retry attempts to $($RetryAttempts)" -Level Verbose 
+                    $JobStep.RetryAttempts = $RetryAttempts
+                }
+
+                if ($RetryInterval) {
+                    Write-Message -Message "Setting job step retry interval to $($RetryInterval)" -Level Verbose 
+                    $JobStep.RetryInterval = $RetryInterval
+                }
+
+                if ($OutputFileName) {
+                    Write-Message -Message "Setting job step output file name to $OutputFileName" -Level Verbose 
+                    $JobStep.OutputFileName = $OutputFileName
```

Comment:
The reason I did not implement that one is because the documentation of the SMO said this:
> The JobStepFlags property is reserved for future use. 

It works though and it was not that difficult to implement.